kiwi/storage/subformat/vagrant_base.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) 2019 SUSE Linux GmbH.  All rights reserved.
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import json
+import os
+import random
+from textwrap import dedent
+from tempfile import mkdtemp
+from string import Template
+from typing import Dict, Optional  # pylint: disable=unused-import
+
+from kiwi.storage.subformat.base import DiskFormatBase
+from kiwi.command import Command
+from kiwi.system.result import Result  # pylint: disable=unused-import
+
+from kiwi.exceptions import (
+    KiwiFormatSetupError
+)
+
+
+def get_vagrantfile_template(custom_settings=None):
+    # type: (Optional[str]) -> Template
+    """
+    **Generate a Vagrantfile configuration template**
+
+    This function creates a simple template for the Vagrantfile that is
+    included inside a vagrant box.
+
+    The included Vagrantfile carries additional information for vagrant: by
+    default that is just the boxes' MAC address, but depending on the provider
+    additional information need to be present. These can be passed as the
+    parameter ``custom_settings``.
+
+    :param custom_settings: String of additional settings that get pasted into
+        the Vagrantfile template. The string is put at the correct indentation
+        level for you, but the internal indentation has to be provided by the
+        caller.
+    :type custom_settings: str.
+    :returns: A :class:`string.Template` with ``custom_settings`` inserted at
+        the appropriate position. The template has one the variable
+        ``mac_address`` that must be substituted.
+
+    :Example:
+
+    The default without any additional settings will result in this
+    Vagrantfile:
+    >>> print(get_vagrantfile_template().substitute({'mac_address': 'deadbeef'}))
+    Vagrant.configure("2") do |config|
+      config.vm.base_mac = "deadbeef"
+    end
+
+    If your provider/box requires additional settings, provide them as follows:
+    >>> extra_settings = dedent('''
+    ... config.vm.hostname = "no-dead-beef"
+    ... config.vm.provider :special do |special|
+    ...   special.secret_settings = "please_work"
+    ... end
+    ... ''').strip()
+    >>> print(get_vagrantfile_template(extra_settings).substitute({'mac_address': 'DEADBEEF'}))
```

Comment:
That information imho belongs to a template class specification. For coding examples we use

```python
"""
.. code:: python
     ...
"""
```

The :returns: as well as :Example: as well as :type: seems to be not evaulated by the sphinx docs. So for method doc strings we use this format all over the place

```
def foo(arg):
    """
    Does something

    :param type arg: arg for foo

    :return: whatever it returns

    :rtype: return_type
    """
```

Could you adapt that to be consistent with the rest of the code base ? I know there are many ways to do this but it wold be good to have a common schema

Reply:
Will do that!

Out of curiosity: do you know if sphinx can actually run these python code examples? I have added this as a doctest because they can be tested relatively easily via the doctest module.